client/pages/auth.jsx

```diff
@@ -15,36 +30,94 @@ function AuthForm({ existingUsernames }) {
 
   function handleSubmit(e) {
     e.preventDefault();
+    const authForm = document.querySelector('.auth-form');
+    authForm.reset();
     const { username } = userInfo;
-    if (existingUsernames.includes(username)) setDisplayMessage('error');
-    fetch('/api/sign-up', {
-      method: 'POST',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify(userInfo)
-    })
-      .catch(err => console.error('ERROR:', err));
-    setDisplayMessage('success');
-    setTimeout(() => {
-      window.location.hash = 'sign-in';
-    }, 4000);
+    if (action.type === 'sign-up') {
+      if (existingUsernames.includes(username)) setAction({ ...action, message: 'error' });
+      fetch('/api/auth/sign-up', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(userInfo)
+      })
+        .then(response => response.json())
+        .then(result => setAction({ ...action, message: 'success' }))
+        .catch(err => console.error('ERROR:', err));
+      setTimeout(() => {
+        window.location.hash = 'sign-in';
+      }, 2000);
```

Comment:
Using a timeout like this is hacky. Instead move the hash change to your second `.then()` callback.